### ISSUE TITLE
Reproduce transformer on Zh-EN corpus 

### ISSUE THREAD

**Issue by KelleyYin** [pytorch/fairseq#250]
Hi,
#233 have reproduced the results in WMT corpus. Therefore, I basically simulated his hyper-parameter setup for training Zh-EN corpus.
However, the Zh-EN corpus is much less than WMT, only have 1.25 millions . In addition, we can't joint dictionary for Zh-En, and I only available 3 Titan Xp GPUs for training.

My conmands as followed:
`CUDA_VISIBLE_DEVICES=3,4,5 python train.py -data $DATA_RAW/ch-en --raw-text -s ch -t en --lr 0.0001 --min-lr 1e-09      --weight-decay 0.0 --clip-norm 0.0 --dropout 0.3 --max-tokens 4300 --arch transformer --optimizer adam --adam-betas     '(0.9, 0.98)' --lr-scheduler inverse_sqrt --warmup-init-lr 1e-07 --warmup-updates 4000 --save-dir checkpoints/transformer-acc --criterion label_smoothed_cross_entropy --label-smoothing 0.1 --update-freq 42
`

However, my model converges very slowly. I wonder whether my command is correct .

The training log is 
`| epoch 001:   1000 / 2969 loss=15.483, nll_loss=15.471, ppl=45403.52, acc=0, wps=28586, ups=0.1, wpb=506004, bsz=17    838, num_updates=23, lr=6.74425e-07, gnorm=0.137, clip=100%, oom=0, wall=407
173 | epoch 001:   2000 / 2969 loss=15.298, nll_loss=15.266, ppl=39392.97, acc=0.02, wps=36297, ups=0.1, wpb=506167, bsz    =17798, num_updates=47, lr=1.27383e-06, gnorm=0.133, clip=100%, oom=0, wall=655
174 | epoch 001 | loss 15.052 | nll_loss 14.992 | ppl 32580.35 | acc 1.14 | wps 39362 | ups 0.1 | wpb 505912 | bsz 17741     | num_updates 70 | lr 1.84825e-06 | gnorm 0.126 | clip 100% | oom 0 | wall 900
175 | epoch 001 | valid on 'valid' subset | valid_loss 13.772 | acc 5.51 | valid_nll_loss 13.5649 | valid_ppl 12118.07 |     num_updates 70
176 | epoch 002:   1000 / 2969 loss=14.002, nll_loss=13.824, ppl=14501.08, acc=6.51, wps=47083, ups=0.1, wpb=520937, bsz    =18226, num_updates=93, lr=2.42268e-06, gnorm=0.115, clip=100%, oom=0, wall=1154
177 | epoch 002:   2000 / 2969 loss=13.795, nll_loss=13.592, ppl=12344.16, acc=6.46, wps=47998, ups=0.1, wpb=513373, bsz    =18036, num_updates=117, lr=3.02207e-06, gnorm=0.104, clip=100%, oom=0, wall=1402
178 | epoch 002 | loss 13.635 | nll_loss 13.413 | ppl 10903.88 | acc 6.39 | wps 47885 | ups 0.1 | wpb 510854 | bsz 17903     | num_updates 140 | lr 3.5965e-06 | gnorm 0.096 | clip 100% | oom 0 | wall 1646
179 | epoch 002 | valid on 'valid' subset | valid_loss 12.6701 | acc 5.51 | valid_nll_loss 12.3103 | valid_ppl 5078.84 |     num_updates 140 | best 12.6701
180 | epoch 003:   1000 / 2969 loss=13.103, nll_loss=12.817, ppl=7217.25, acc=6.18, wps=34308, ups=0.1, wpb=520518, bsz=    18237, num_updates=163, lr=4.17093e-06, gnorm=0.089, clip=100%, oom=0, wall=1995
181 | epoch 003:   2000 / 2969 loss=13.019, nll_loss=12.723, ppl=6761.40, acc=6.15, wps=40391, ups=0.1, wpb=512780, bsz=    17888, num_updates=187, lr=4.77032e-06, gnorm=0.084, clip=100%, oom=0, wall=2243
182 | epoch 003 | loss 12.951 | nll_loss 12.646 | ppl 6411.05 | acc 6.14 | wps 42484 | ups 0.1 | wpb 510901 | bsz 17884     | num_updates 210 | lr 5.34475e-06 | gnorm 0.079 | clip 100% | oom 0 | wall 2488
183 | epoch 003 | valid on 'valid' subset | valid_loss 12.0541 | acc 5.51 | valid_nll_loss 11.6281 | valid_ppl 3165.19 |     num_updates 210 | best 12.0541
184 | epoch 004:   1000 / 2969 loss=12.679, nll_loss=12.343, ppl=5196.66, acc=6.34, wps=36366, ups=0.1, wpb=521695, bsz=    18180, num_updates=233, lr=5.91917e-06, gnorm=0.075, clip=100%, oom=0, wall=2818
185 | epoch 004:   2000 / 2969 loss=12.622, nll_loss=12.279, ppl=4970.45, acc=6.39, wps=41717, ups=0.1, wpb=513174, bsz=    18040, num_updates=257, lr=6.51857e-06, gnorm=0.071, clip=100%, oom=0, wall=3066
186 | epoch 004 | loss 12.564 | nll_loss 12.214 | ppl 4752.54 | acc 6.43 | wps 43451 | ups 0.1 | wpb 510876 | bsz 17905     | num_updates 280 | lr 7.093e-06 | gnorm 0.069 | clip 100% | oom 0 | wall 3311
187 | epoch 004 | valid on 'valid' subset | valid_loss 11.6169 | acc 5.73 | valid_nll_loss 11.1444 | valid_ppl 2263.61 |     num_updates 280 | best 11.6169

.........

| epoch 046:   1000 / 2969 loss=7.891, nll_loss=6.795, ppl=111.03, acc=27.6, wps=43100, ups=0.1, wpb=522338, bsz=18569, num_updates=3173, lr=7.93457e-05, gnorm=0.034, clip=100%, oom=0, wall=36610
| epoch 046:   2000 / 2969 loss=7.890, nll_loss=6.793, ppl=110.90, acc=27.64, wps=45802, ups=0.1, wpb=513610, bsz=18141, num_updates=3197, lr=7.99451e-05, gnorm=0.034, clip=100%, oom=0, wall=36858
| epoch 046 | loss 7.893 | nll_loss 6.797 | ppl 111.19 | acc 27.56 | wps 46350 | ups 0.1 | wpb 510919 | bsz 17904 | num_updates 3220 | lr 8.05195e-05 | gnorm 0.034 | clip 100% | oom 0 | wall 37102
| epoch 046 | valid on 'valid' subset | valid_loss 8.05595 | acc 22.88 | valid_nll_loss 6.95224 | valid_ppl 123.83 | num_updates 3220 | best 8.03743
| epoch 047:   1000 / 2969 loss=7.857, nll_loss=6.757, ppl=108.15, acc=27.76, wps=39142, ups=0.1, wpb=520066, bsz=17893, num_updates=3243, lr=8.10939e-05, gnorm=0.034, clip=100%, oom=0, wall=37408
| epoch 047:   2000 / 2969 loss=7.847, nll_loss=6.744, ppl=107.17, acc=27.83, wps=43565, ups=0.1, wpb=513354, bsz=17869, num_updates=3267, lr=8.16933e-05, gnorm=0.034, clip=100%, oom=0, wall=37656
| epoch 047 | loss 7.843 | nll_loss 6.739 | ppl 106.82 | acc 27.87 | wps 44753 | ups 0.1 | wpb 510888 | bsz 17905 | num_updates 3290 | lr 8.22678e-05 | gnorm 0.034 | clip 100% | oom 0 | wall 37902
| epoch 047 | valid on 'valid' subset | valid_loss 8.01615 | acc 22.91 | valid_nll_loss 6.90708 | valid_ppl 120.02 | num_updates 3290 | best 8.01615
| epoch 048:   1000 / 2969 loss=7.813, nll_loss=6.705, ppl=104.32, acc=28.01, wps=40917, ups=0.1, wpb=520265, bsz=18069, num_updates=3313, lr=8.28422e-05, gnorm=0.034, clip=100%, oom=0, wall=38194
| epoch 048:   2000 / 2969 loss=7.810, nll_loss=6.701, ppl=104.06, acc=28.01, wps=44627, ups=0.1, wpb=513564, bsz=18005, num_updates=3337, lr=8.34416e-05, gnorm=0.034, clip=100%, oom=0, wall=38442

`

**Comment by KelleyYin** [comment]
It's weird . 
I changed the learning rate to 0.001 and the other parameters are the same as above .  The model doesn't achieve best performence at 256 epochs , which took two days .

So I also update `--update-freq` to one and used 3-gpus .  The best results were achieved at 40 epochs after 12 hours .  The multi-bleu score as follow:

| nist02 | nist03 |  nist04 | nist05 | nist08 |
| ------ | ------ | ------ |------ |------ |
|  44.93 | 43.36  | 45.13  | 43.78  | 34.90  |

However, I used only one gpu with the same hyper-parameters for training . The performance is also terrible!
